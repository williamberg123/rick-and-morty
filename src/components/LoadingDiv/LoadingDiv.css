.LoadingDiv {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 70px;
}

.dots {
    width: 20px;
    height: 20px;
    border-radius: 50%;
    background-color: rgb(114, 114, 114);
    margin: 0 5px;
    animation: loading 1s infinite;
}

.LoadingDiv .dots:nth-child(2) {
    animation-delay: 0.1225s;
}

.LoadingDiv .dots:nth-child(3) {
    animation-delay: 0.25s;
}

.LoadingDiv .dots:nth-child(4) {
    animation-delay: 0.375s;
}

.LoadingDiv .dots:nth-child(5) {
    animation-delay: 0.5s;
}

@media (max-width: 450px) {
    .dots {
        width: 10px;
        height: 10px;
        margin: 0 4px;
    }
}

@keyframes loading {
    0% {
        transform: translateY(0px);
    }

    50% {
        transform: translateY(-10px);
    }

    100% {
        transform: translateY(0px);
    }
}