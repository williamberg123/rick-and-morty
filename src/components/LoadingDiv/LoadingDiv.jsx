import React from 'react';

import './LoadingDiv.css';

export default function LoadingDiv(){
    return (
        <div className="LoadingDiv" data-testid="LoadingDiv">
            <div className="dots"></div>
            <div className="dots"></div>
            <div className="dots"></div>
            <div className="dots"></div>
            <div className="dots"></div>
        </div>
    );
}