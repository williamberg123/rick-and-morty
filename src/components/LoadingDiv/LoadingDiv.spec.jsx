import React from 'react';
import { screen, render } from '@testing-library/react';
import LoadingDiv from './LoadingDiv';

describe('<LoadingDiv />', () => {
    it('should render all dots', () => {
        render(<LoadingDiv />);

        expect(screen.getByTestId('LoadingDiv').getElementsByClassName('dots'))
            .toHaveLength(5);
    });

    it('should match snapshot', () => {
        render(<LoadingDiv />);

        expect(screen.getByTestId('LoadingDiv')).toMatchSnapshot();
    });
});

// Testado