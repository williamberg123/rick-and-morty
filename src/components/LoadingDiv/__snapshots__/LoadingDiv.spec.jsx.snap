// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingDiv /> should match snapshot 1`] = `
<div
  class="LoadingDiv"
  data-testid="LoadingDiv"
>
  <div
    class="dots"
  />
  <div
    class="dots"
  />
  <div
    class="dots"
  />
  <div
    class="dots"
  />
  <div
    class="dots"
  />
</div>
`;
